### PR TITLE
Changed how the bot handles terminal commands

### PR DESCRIPTION
**Pull request by Heniboj**
If there is a terminal open the bot will check every message and deletes the terminal commands since you can see them in the terminal window too.

### CODE REVIEW

**Comment by Heniboj** [comment]
This pull request would also fix issue #17

**Review by Adikso — CHANGES_REQUESTED** [review]
Hey, thanks for contributing to the project 🎉
From what I can see your code is never going to be executed. In this else-branch there are processed only commands without dot after a prefix - commands for bash, not for bashbot. Bashbot commands are processed in the previous if-branch.

What is more bashbot has a configurable prefix, so we can't hardcode checking the dollar symbol.

**Comment by Heniboj** [comment]
My intention was to remove the commands that will be executed in the command prompt, since if you send a discord message it would push the command prompt away and you always had to make the $.here command and you see your past commands in there anyway. Checking for the dot is necessary because otherwise the bot would delete non-bash commands. And I checked for the dollar sign not to accidentally delete regular discord messages. Since you said that the prefix could be changed, I could make the bot recognize that if you want.

File: bashbot/bot.py [hunk 1]
```diff
@@ -65,7 +65,10 @@ async def on_message(self, message: Message):
 
         if self.is_invoke(message):
```
Comment (Adikso) [inline bashbot/bot.py:66]: This is checking if the message is starting with prefix + dot so inside "elif" you will never have a message with prefix + dot.

File: bashbot/bot.py [hunk 2]
```diff
@@ -65,7 +65,10 @@ async def on_message(self, message: Message):
...
             await self.process_commands(message)
+
         elif terminal and terminal.state == TerminalState.OPEN:
+            if(message.content[0] == "$" and message.content[1] != "."):
```
Comment (Adikso) [inline bashbot/bot.py:70]: In consequence, this condition will never resolve to "true"

File: bashbot/bot.py (hunk 1 above)
Comment (Adikso) [inline bashbot/bot.py:66]: This if-condition is checking if message is starting with prefix + dot. So it is not possible to have a message starting with prefix + dot in elif branch.

**Comment by Adikso** [comment]
Keep in mind that the message should also be deleted when terminal is in "interactive mode". This mode does not require user to put a prefix before the command.

**Pull request closed without merging**